test/files/pos/t5183.scala

```diff
@@ -0,0 +1,7 @@
+trait Day
+
+object Test {
+  def foo(t: Int with Day) = t == t
+}
+
+class DayOps(val i: Int with Day) extends AnyVal
```

Comment:
could you include the other / related examples that are mentioned in the various comments in the ticket? (case classes with 1 and 2 fields, the varargs issue from david's gist).